### PR TITLE
Added @listen_to decorator + tests

### PR DESCRIPTION
**Pull request by gpailler**
- Added new decorator @listen_to to listen all messages on subscribed channels
- Added reply_webapi() and send_webapi() functions to use Slack Web API (allow messages formatting when using bot integration)
- Added support for bot_icon and bot_emoji (used by Web API)
- Allowed multiples plugins to be called for the same message

### CODE REVIEW

**Comment by gpailler** [comment]
Tests are here: https://travis-ci.org/gpailler/slackbot/builds/58080039
Travis is unable to run tests for this PR because some env variables are hidden

File: tests/functional/test_functional.py [hunk 1]
```diff
@@ -80,19 +88,35 @@ def test_bot_upload_file_from_link(driver):
...
-def test_bot_ignores_non_related_channel_message(driver):
+def test_bot_ignores_non_related_message_response_tosender(driver):
+    driver.send_channel_message('hello', tobot=True)
+    driver.ensure_only_specificmessage_from_bot('hello sender!', tosender=True)
+
+def test_bot_ignores_non_related_message_response_tochannel(driver):
     driver.send_channel_message('hello', tobot=False)
-    driver.ensure_no_channel_reply_from_bot()
+    driver.ensure_only_specificmessage_from_bot('hello channel!', tosender=False)
+
```
Comment (lins05) [inline tests/functional/test_functional.py:112]: Here we also need tests that checks "the bot must not send any reply if the channel message doesn't match any registered plugin". I think I have written that tests, but seems you have overwritten it?

File: slackbot/bot.py [hunk 2]
```diff
@@ -19,7 +19,11 @@
 
 class Bot(object):
     def __init__(self):
-        self._client = SlackClient(settings.API_TOKEN)
+        self._client = SlackClient(
+            settings.API_TOKEN,
+            bot_icon = settings.BOT_ICON if hasattr(settings, 'BOT_ICON') else None,
```
Comment (lins05) [inline slackbot/bot.py:24]: Please, follow PEP8 as much as possible. There should not be any space before/after the equal character of keyward arguments.

File: slackbot/bot.py [hunk 3]
```diff
@@ -66,16 +73,27 @@ def _load_plugins(self, plugin):
             except:
                 logger.exception('Failed to import %s', module)
 
-    def get_plugin(self, text):
-        for matcher in self.commands:
-            m = matcher.match(text)
+    def get_plugins(self, category, text):
+        hasMatch = False
```
Comment (lins05) [inline slackbot/bot.py]: again, please use `under_score` instead of `camelCase`.

**Comment by lins05** [comment]
:+1:  This PR mostly looks good to me. See my comments about tests and code styles. I think we should also update REAMDE about the `listen_to` decorator? (Acutally when I add a new feature I tend to update README before writing any code).

File: slackbot/bot.py (hunk 2 above)
Comment (gpailler) [inline slackbot/bot.py:24]: Done. I updated the code to follow PEP8 guidelines (except for lines length)

File: slackbot/bot.py (hunk 3 above)
Comment (gpailler) [inline slackbot/bot.py]: Done

File: tests/functional/test_functional.py (hunk 1 above)
Comment (gpailler) [inline tests/functional/test_functional.py:112]: Test re-added + another

**Comment by gpailler** [comment]
I think I applied all your comments + updated README.

New tests result: https://travis-ci.org/gpailler/slackbot/builds/59854080

**Comment by lins05** [comment]
Thanks! I'll make a new release soon.

**Pull request merged**